src/schema/schema_alter.c

```diff
@@ -72,10 +73,12 @@ __alter_file(WT_SESSION_IMPL *session, const char *newcfg[])
      * it and the next open will see the updated metadata.
      */
     uri = session->dhandle->name;
-    if (!WT_PREFIX_MATCH(uri, "file:"))
-        return (__wt_unexpected_object_type(session, uri, "file:"));
+    if (!WT_PREFIX_MATCH(uri, "file:") && !WT_PREFIX_MATCH(uri, "tier:"))
```

Comment:
@sueloverso after this change do you think I should rename the method from `__alter_file` to `__alter_source` or something like that?

Reply:
`alter_tiered` now calls `__schema_alter` for `file:`, `tier:` and `object:`.